lib/nerves_hub_link/socket.ex

```diff
@@ -447,6 +453,33 @@ defmodule NervesHubLink.Socket do
     {:noreply, socket}
   end
 
+  defp network_connected?(%{assigns: %{config: config}}) do
+    if config.wait_for_network do
+      host = to_charlist(config.device_api_host)
+
+      case :inet.gethostbyname(host) do
+        {:ok, _} -> true
+        _ -> false
+      end
+    else
+      true
+    end
+  end
+
+  defp time_synchronized?(%{assigns: %{config: config}}) do
+    if config.wait_for_time_sync do
+      NervesTime.synchronized?()
+    else
+      true
+    end
+  end
+
+  defp network_check_with_message(msg, socket) do
```

Comment:
```suggestion
  defp log_and_reschedule_checks(msg, socket) do
```
What do you think of a more purpose built function name?

Reply:
your naming is much better